backend/app.js

```diff
@@ -31,7 +32,7 @@ botFactory.makeBot(process.env.FOOSBOT_TOKEN, process.env.FOOS_CHANNEL_NAME)
     })
     .catch((error) => console.log('Bot initialization failed', error))
 
-
+const matchReporter = new MatchReporter(process.env.MATCH_RESULT_PREFIX_SUFFIX_CONFIG)
```

Comment:
thinking - what if match reporter also took the bot as a constructor?

Reply:
`bot` is still `undefined` at the time the constructor is called, so it's not that simple. But what you asked for makes perfect sense... So I am now passing all the env. variables to the constructor and initializing the bot there. Plus I took the refactoring even furhter... I'm gonna push it soon. :)